service_discovery/core/src/service_registry_impl.rs

```diff
@@ -120,52 +111,38 @@ impl ServiceRegistry for ServiceRegistryImpl {
 
     /// Discovers a list of services based on the namespace, or logical grouping of services.
     ///
-    /// This function discovers a list of services based on a [`DiscoverByNamespaceRequest`]. Returns a
-    /// [`DiscoverByNamespaceResponse`].
-    ///
     /// # Arguments
     ///
-    /// * `request` - A [`DiscoverByNamespaceRequest`] wrapped by a [`tonic::Request`].
+    /// * `request` - Contains the namespace of the services to be discovered
     async fn discover_by_namespace(
         &self,
         request: Request<DiscoverByNamespaceRequest>,
     ) -> Result<Response<DiscoverByNamespaceResponse>, Status> {
         let namespace = request.into_inner().namespace;
-        let service_list: Vec<ServiceMetadata>;
 
         // This block controls the lifetime of the lock.
-        {
-            service_list = {
-                let lock = self.registry_map.read();
-                lock.iter()
-                    .filter_map(|(service_identifier, service_metadata)| {
-                        if service_identifier.namespace == namespace {
-                            Some(service_metadata.clone())
-                        } else {
-                            None
-                        }
-                    })
-                    .collect()
-            };
-        }
-        if service_list.is_empty() {
-            Err(Status::not_found(format!("No registrations found for namespace {namespace}")))
-        } else {
-            let discover_by_namespace_response =
-                DiscoverByNamespaceResponse { services: service_list };
-            Ok(Response::new(discover_by_namespace_response))
-        }
+        let service_list: Vec<ServiceMetadata> = {
+            let lock = self.registry_map.read();
+            lock.iter()
+                .filter_map(|(service_identifier, service_metadata)| {
+                    if service_identifier.namespace == namespace {
```

Comment:
looks like this could be shortened to a more idiomatic [bool.then](https://doc.rust-lang.org/std/primitive.bool.html#method.then):

```rust
(service_identifier.namespace == namespace).then(|| service_metadata.clone())```

Reply:
Thanks, I wasn't aware of this. Updated